src/modules/fancyzones/lib/ZoneWindowDrawing.cpp

```diff
@@ -106,22 +110,22 @@ ZoneWindowDrawing::ZoneWindowDrawing(HWND window)
     m_renderThread = std::thread([this]() { RenderLoop(); });
 }
 
-void ZoneWindowDrawing::Render()
+ZoneWindowDrawing::RenderResult ZoneWindowDrawing::Render()
 {
     std::unique_lock lock(m_mutex);
 
     if (!m_renderTarget)
     {
-        return;
+        return RenderResult::Failed;
```

Comment:
Do we need to return `RenderResult::Failed` instead of just `RenderResult::AnimationEnded`?
Also the caller is not currently handling `RenderResult::Failed` .

Reply:
No, `RenderResult::AnimationEnded` has a specific meaning, it should tell the render loop to hide the window. `RenderResult::Failed` should tell it that an error occurred and rendering is disabled, because the render target couldn't be created. For now we just ignore it but we can add code to process such errors later if needed.